draft-gpew-priv-ppm.md

```diff
@@ -672,247 +677,375 @@ independently.
 Multiple aggregation flows can be executed at the same time."}
 
-[OPEN ISSUE: Should there be an indication of whether a given AggregateReq is a
+[OPEN ISSUE: Should there be an indication of whether a given aggregate request is a
 continuation of a previous sub-batch?]
 
-[TODO: Decide if and how the collector's request is authenticated.]
+The aggregation flow can be thought of as having three phases for transforming
+each valid input report share into an output share:
 
+- Initialization: Begin the aggregation flow by sharing report shares with each
+  helper. Each aggregator, including the leader, initializates the underlying
+  VDAF instance using these report shares and the VDAF configured for the
+  corresponding measurement task.
+- Continuation: Continue the aggregation flow by exchanging messages produced
+  by the underlying VDAF instance until aggregation completes or an error occurs.
+  These messages do not replay the shares.
+- Completion: Finish the aggregate flow, yielding an output share corresponding
+  for each input report share in the batch.
 
-### Aggregate Request
+### Aggregate Initialization {#agg-init}
 
-The AggregateReq request is used by the leader to send a set of reports to the
-helper. These reports MUST all be associated with the same PPM task and batch.
+The leader begins aggregation by choosing a set of candidate reports that pertain
+to the same PPM task. After choosing the set of candidates, the leader begins
+aggregation by splitting each report into "report shares", one for each aggregator.
+The leader and helpers then run the aggregate initialization flow to accomplish two tasks:
 
-For each aggregator endpoint `[aggregator]` in `AggregateReq.task_id`'s
-parameters except its own, the leader sends a POST request to
-`[aggregator]/aggregate` with the following message:
+1. Recover and determine which input report shares are invalid.
+1. For each valid report share, initialize the VDAF preparation process.
+
+An invalid report share is marked with one of the following errors:
+
+~~~
+enum {
+  batch-collected(0),
+  report-replayed(1),
+  report-dropped(2),
+  hpke-unknown-config-id(3),
+  hpke-decrypt-error(4),
+  vdaf-prep-error(5),
+} ReportShareError;
+~~~
+
+#### Leader Initialization
+
+The leader begins the aggregate initialization phase with the set of candidate report
+shares as follows:
+
+1. Decrypt the input share for each report share as described in {{input-share-decryption}}.
+1. Check that the resulting input share is valid as described in {{input-share-batch-validation}}.
+1. Initialize VDAF preparation as described in {{input-share-prep}}.
+
+If any step yields an invalid report share, the leader removes the report share from
+the set of candidate reports. Once the leader has initialized this state for all valid
+candidate report shares, it then creates an AggregateInitReq message for each helper to
+initialize the preparation of this candidate set. This message is structured as follows:
```

Comment:
Does this have to be done serially? Could the leader instead send all of the shares and then discover later that one is bogus? Surely yes, because the proof could fail.

Reply:
Yep, it can. I added a note to that effect.